packages/build-ui/src/pages/agits/_id/management/artists/_id/components/artist_table.rs

```diff
@@ -79,12 +80,20 @@ pub fn ArtistTable(assets: Vec<Assets>     ,lang: Language) -> Element {
                 }
             }
             tbody {
-                { assets.iter().enumerate().map(|(index, asset)| {
+                { assets.into_iter().enumerate().map(|(index, asset)| {
                     let is_dropdown_open = active_dropdown.with(|active| active == &Some(index));
                     rsx! {
                         tr {
                             key: "owned-tr-{index}",
                             class: "border-b border-gray-800 hover:bg-gray-900",
+                            onclick: move |_| {
+                             
+                                use_navigator().push(Route::EditArtistPage { 
+                                    lang: lang, 
+                                    agit_id: agit_id(), 
+                                    artist_id: artist_id
+                                });
+                            },
```

Comment:
If you handle the behavior within the Component like this, scalability decreases.
Instead of having the Component call the navigator to navigate, it would be good to receive `onclick` as props and let the caller decide the action.
```suggestion
                            onclick: move |_| {
                                on_asset_click.call(agit_id, artist_id);
                            },
```